/*
 * Read the following code and check whether there is anything wrong with it. Then write some similar code and
 * check your answer.
 * 1
 * // Some code here
 * try {
 *   // more code here
 *   list.add(newElement);
 *   // more code here
 * } catch (Exception ex) {
 *   ex.printStackTrace();
 * } catch (NullPointerException ex) {
 *   ex.printStackTrace();
 * }
 */
 
 The NullPointerException ex should be before the Exception ex in the catch blocks
 as the general Exception ex will catch any NullPointerExceptions before the code
 can get to the NullPointerException catch block, rendering it pointless.
 
 1
 // Some code here
 try {
   // more code here
   list.add(newElement);
   // more code here
 } catch (NullPointerException ex) {
   ex.printStackTrace();
 } catch (Exception ex) {
   ex.printStackTrace();
 }
 
 Then if a NullPointerException happens it will be indicated, with any other exceptions being 
 caught by the second Exception ex.
 